src/Common/Core/Impl/Logging/Implementation/LoggingPermissions.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.Common.Core.Extensions;
+using Microsoft.Common.Core.OS;
+using Microsoft.Common.Core.Shell;
+using Microsoft.Common.Core.Telemetry;
+using Microsoft.Win32;
+
+namespace Microsoft.Common.Core.Logging {
+    /// <summary>
+    /// Determines logging permissions depending on the current 
+    /// telemetry settings and registry keys.
+    /// </summary>
+    /// <remarks>
+    /// Rules:
+    ///     a. Feedback and logging is permitted if telemetry is enabled. If telemetry is off, 
+    ///        admin can selectively enable feedback and logging in registry under HKLM.
+    ///     b. If telemetry is on, default logging level is 'Normal'. This allows recording of all events
+    ///        but security and privacy related data such as cross-process traffic or commands and data 
+    ///        in the interactive window.
+    ///     c. Admin can limit level of logging and disable feedback sending even if app telemetry is on.
+    /// </remarks>
+    [Export(typeof(ILoggingPermissions))]
+    internal sealed class LoggingPermissions : ILoggingPermissions {
```

Comment:
It has been marked as MEF container in the manifest - I think since we had IFileSystem exported back then